### DISCUSSION TITLE
Login with Public REST API

### DISCUSSION THREAD

**Discussion by ghost** [Budibase/budibase#4933] [General]
I am really excited about the new Public Rest API. One challenge I see however is that there is no Login or Authenticate method that allows for passing User Credentials and getting back the user API key? I also don't see anyway for Triggers to retrieve the API key (for example, to be able to email it to the user rather than having them have to login and use the Account UI.

Is the public Rest API only meant to be used for server-to-server integrations? I was sort of hoping that I could use it to integrate a client application as well.

**Comment by mjashanks** [comment]
If you want to use it for client applications using a standard login process, then you can do so by using cookies...

1. Your login forms sends username/password
2. Server returns a response and a `Set-Cookie` header
3. Set your `Cookie` to that returned value
4. Continue using the API, without the API key

The purpose of the API Key is so we can skip the whole login process.
